webknossos/webknossos/dataset/remote_dataset_registry.py

```diff
@@ -0,0 +1,48 @@
+from typing import TYPE_CHECKING, Optional, Sequence, TypeVar, Union
+
+from webknossos.utils import LazyReadOnlyDict
+
+if TYPE_CHECKING:
+    from webknossos.dataset.dataset import RemoteDataset
+
+
+K = TypeVar("K")  # key
+V = TypeVar("V")  # value
+C = TypeVar("C")  # cache
+
+
+class RemoteDatasetRegistry(LazyReadOnlyDict[str, "RemoteDataset"]):
+    """Dict-like class mapping dataset names to `RemoteDataset` instances."""
+
+    def __init__(
+        self,
+        organization_id: Optional[str],
+        tags: Optional[Union[str, Sequence[str]]],
+    ) -> None:
+        from webknossos.administration.user import User
+        from webknossos.client._generated.api.default import dataset_list
+        from webknossos.client.context import _get_generated_client
+        from webknossos.dataset.dataset import Dataset
+
+        client = _get_generated_client(enforce_auth=True)
+
+        if organization_id is None:
+            organization_id = User.get_current_user().organization_id
```

Comment:
The code for filtering by organization is missing.

Reply:
Good catch! :see_no_evil: Fixed it in the new commit.